js/pagedtable.js

```diff
@@ -736,131 +786,274 @@ var PagedTable = function (pagedTable, source) {
-    thead.innerHTML = "";
-  };
-
-  var renderHeader = function(clear) {
-    cachedPagedTableClientWidth = pagedTable.clientWidth;
-
-    var fragment = document.createDocumentFragment();
-
-    header = document.createElement("tr");
-    fragment.appendChild(header);
-
-    if (columns.number > 0)
-      header.appendChild(renderColumnNavigation(-columns.visible, true));
-
-    columns.subset = columns.subset.map(function(columnData) {
-      var column = document.createElement("th");
-      if (typeof(columnData.align) === "undefined") columnData.align = "left";
-      column.setAttribute("align", columnData.align);
-      column.style.textAlign = columnData.align;
+    var columnName = document.createElement("div");
+    columnName.setAttribute("class", "pagedtable-header-name");
+    if (columnData.label === "") {
+      columnName.innerHTML = "&nbsp;";
+    } else {
+      columnName.appendChild(document.createTextNode(columnData.label));
+    }
+    column.appendChild(columnName);
 
-      column.style.maxWidth = maxColumnWidth(null);
-      if (columnData.width) {
-        column.style.minWidth =
-          column.style.maxWidth = maxColumnWidth(columnData.width);
+    var columnType = document.createElement("div");
+    columnType.setAttribute("class", "pagedtable-header-type");
+    if (columnData.type === "") {
+      columnType.innerHTML = "&nbsp;";
+    } else {
+      if (typeof(columnData.type) !== "undefined") {
+        columnType.appendChild(document.createTextNode("<" + columnData.type + ">"));
       }
+    }
+    column.appendChild(columnType)
+    return(column);
+  };
+  
+  var makeColumnBodyElements = function(idx){
+    
+    var columnData = columns.subset[idx]
 
-      var columnName = document.createElement("div");
-      columnName.setAttribute("class", "pagedtable-header-name");
-      if (columnData.label === "") {
-        columnName.innerHTML = "&nbsp;";
-      }
-      else {
-        columnName.appendChild(document.createTextNode(columnData.label));
+    var fragment = document.createDocumentFragment();
+    
+    // assumption that all fields have the same keys? valid?
+    var idx_name = columnData.name
+    
+    var column_elements = []
+    
+    data.forEach(function(dataRow,idxRow) {
+      
+      var dataCell = dataRow[idx_name];
+      var htmlCell = document.createElement("td");
+      htmlCell.setAttribute("class", "col_" + idx)
+      htmlCell.style.display = "&nbsp";
+
+
+      if (typeof(dataCell) !== "string") dataCell = dataCell.toString();
+      if (dataCell === "NA") htmlCell.setAttribute("class", "pagedtable-na-cell");
+      if (dataCell === "__NA__") dataCell = "NA";
+
+      var cellText = document.createElement('div');
+      cellText.style.width = "fit-content"
+      
+      //revert for REALSIES // ************************************************************************
+      //cellText.innerHTML = makeCellContents(dataCell);
```

Comment:
This is the main function, so we should enable it.

```suggestion
      cellText.innerHTML = makeCellContents(dataCell);
```

Reply:
I was getting weird errors when I was testing, It would run fine but breakpoints it would throw an error within dompurify, so for testing I paused it.